lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/schema/JsonSchemaDaoImplTest.java

```diff
@@ -318,4 +308,532 @@ public void testGetVersionInfoNullId() {
+		JsonSchemaVersionInfo expected = createNewSchemaVersion("my.org.edu/foo.bar/1.0.1", index++);
+		createNewSchemaVersion("my.org.edu/foo.bar", index++);
+		createNewSchemaVersion("other.org/foo.bar", index++);
+		createNewSchemaVersion("my.org.edu/foo.bar.other", index++);
+
+		// call under test
+		JsonSchemaVersionInfo resultInfo = jsonSchemaDao.getVersionInfo("my.org.edu", "foo.bar", "1.0.1");
+		assertEquals(expected, resultInfo);
+	}
+
+	@Test
+	public void testGetSchemaVersionOrganizationNotFound() throws JSONObjectAdapterException {
+		int index = 0;
+		createNewSchemaVersion("my.org.edu/foo.bar/1.0.1", index++);
+		String message = assertThrows(NotFoundException.class, () -> {
+			jsonSchemaDao.getVersionInfo("other.org", "foo.bar", "1.0.1");
+		}).getMessage();
+		assertEquals(
+				"JSON Schema not found for organizationName: 'other.org' and schemaName: 'foo.bar' and semanticVersion: '1.0.1'",
+				message);
+	}
+
+	@Test
+	public void testGetSchemaVersionSchemaNameNotFound() throws JSONObjectAdapterException {
+		int index = 0;
+		createNewSchemaVersion("my.org.edu/foo.bar/1.0.1", index++);
+		assertThrows(NotFoundException.class, () -> {
+			jsonSchemaDao.getVersionInfo("my.org.edu", "foo.bar.bar", "1.0.1");
+		});
+	}
+
+	@Test
+	public void testGetVersionInfoWithVersionNotFound() throws JSONObjectAdapterException {
+		int index = 0;
+		createNewSchemaVersion("my.org.edu/foo.bar/1.0.1", index++);
+		assertThrows(NotFoundException.class, () -> {
+			jsonSchemaDao.getVersionInfo("my.org.edu", "foo.bar", "1.0.2");
+		});
+	}
+
+	@Test
+	public void testGetVersionInfoWithVersionNullOrganization() throws JSONObjectAdapterException {
+		String organizationName = null;
+		String schemaName = "other.schema.name";
+		String semanticVersion = "1.0.2";
+		assertThrows(IllegalArgumentException.class, () -> {
+			jsonSchemaDao.getVersionInfo(organizationName, schemaName, semanticVersion);
+		});
+	}
+
+	@Test
+	public void testGetVersionInfoWithVersionNullSchema() throws JSONObjectAdapterException {
+		String organizationName = "my.org.edu";
+		String schemaName = null;
+		String semanticVersion = "1.0.2";
+		assertThrows(IllegalArgumentException.class, () -> {
+			jsonSchemaDao.getVersionInfo(organizationName, schemaName, semanticVersion);
+		});
+	}
+
+	@Test
+	public void testGetVersionInfoWithVersionNullVersion() throws JSONObjectAdapterException {
+		String organizationName = "my.org.edu";
+		String schemaName = "other.schema.name";
+		String semanticVersion = null;
+		assertThrows(IllegalArgumentException.class, () -> {
+			jsonSchemaDao.getVersionInfo(organizationName, schemaName, semanticVersion);
+		});
+	}
+
+	@Test
+	public void testDeleteSchema() throws JSONObjectAdapterException {
+		int index = 0;
+		createNewSchemaVersion("my.org.edu/foo.bar/1.0.1", index++);
+		createNewSchemaVersion("my.org.edu/foo.bar/1.0.2", index++);
+		JsonSchemaVersionInfo lastInfo = createNewSchemaVersion("my.org.edu/foo.bar", index++);
```

Comment:
Add in a different schema (org or name) and make sure it does not get deleted (maybe with same semanticVersion)